packages/graphiql/src/components/DocExplorer/TypeDoc.tsx

```diff
@@ -5,207 +5,176 @@
+  const [showDeprecated, setShowDeprecated] = useState(false);
 
-type TypeDocState = {
-  showDeprecated: boolean;
-};
+  const navItem = explorerNavStack[explorerNavStack.length - 1];
+  const type = navItem.def;
 
-export default class TypeDoc extends React.Component<
-  TypeDocProps,
-  TypeDocState
-> {
-  constructor(props: TypeDocProps) {
-    super(props);
-    this.state = { showDeprecated: false };
+  if (!schema || !isType(type)) {
+    return null;
   }
 
-  shouldComponentUpdate(nextProps: TypeDocProps, nextState: TypeDocState) {
-    return (
-      this.props.type !== nextProps.type ||
-      this.props.schema !== nextProps.schema ||
-      this.state.showDeprecated !== nextState.showDeprecated
-    );
+  let typesTitle: string | null = null;
+  let types: readonly (GraphQLObjectType | GraphQLInterfaceType)[] = [];
+  if (type instanceof GraphQLUnionType) {
+    typesTitle = 'possible types';
+    types = schema.getPossibleTypes(type);
+  } else if (type instanceof GraphQLInterfaceType) {
+    typesTitle = 'implementations';
+    types = schema.getPossibleTypes(type);
+  } else if (type instanceof GraphQLObjectType) {
+    typesTitle = 'implements';
+    types = type.getInterfaces();
   }
 
-  render() {
-    const schema = this.props.schema;
-    const type = this.props.type;
-    const onClickType = this.props.onClickType;
-    const onClickField = this.props.onClickField;
-
-    let typesTitle: string | null = null;
-    let types: readonly (GraphQLObjectType | GraphQLInterfaceType)[] = [];
-    if (type instanceof GraphQLUnionType) {
-      typesTitle = 'possible types';
-      types = schema.getPossibleTypes(type);
-    } else if (type instanceof GraphQLInterfaceType) {
-      typesTitle = 'implementations';
-      types = schema.getPossibleTypes(type);
-    } else if (type instanceof GraphQLObjectType) {
-      typesTitle = 'implements';
-      types = type.getInterfaces();
-    }
+  let typesDef;
+  if (types && types.length > 0) {
+    typesDef = (
+      <div id="doc-types" className="doc-category">
+        <div className="doc-category-title">{typesTitle}</div>
+        {types.map(subtype => (
+          <div key={subtype.name} className="doc-category-item">
+            <TypeLink type={subtype} />
+          </div>
+        ))}
+      </div>
+    );
+  }
 
-    let typesDef;
-    if (types && types.length > 0) {
-      typesDef = (
-        <div id="doc-types" className="doc-category">
-          <div className="doc-category-title">{typesTitle}</div>
-          {types.map(subtype => (
-            <div key={subtype.name} className="doc-category-item">
-              <TypeLink type={subtype} onClick={onClickType} />
-            </div>
+  // InputObject and Object
+  let fieldsDef;
+  let deprecatedFieldsDef;
+  if (type && 'getFields' in type) {
+    const fieldMap = type.getFields();
+    const fields = Object.keys(fieldMap).map(name => fieldMap[name]);
```

Comment:
Object.values()

Reply:
Same as above. It would actually work, but typescript would infer `any` as the return type of `Object.values(fieldMap)`